### PR TITLE
Fix bug CTAS with multi partition desc

### LINKED ISSUE #5500: CTAS with multi partition desc error
### Steps to reproduce the behavior (Required)
1. create table bb 
    -> PARTITION BY RANGE(`k1`)(
    ->     START ("2021-03-01") END ("2022-03-31") EVERY (INTERVAL date() MONTH)
    -> )
    -> DISTRIBUTED BY HASH(`dt`) BUCKETS 8
    -> AS  select kt,dt from aa

### Expected behavior (Required)
ok
### Real behavior (Required)
java.lang.NullPointerException: null
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.analyze(Analyzer.java:48) ~[classes/:?]
        at com.starrocks.sql.analyzer.Analyzer.analyze(Analyzer.java:43) ~[classes/:?]
        at com.starrocks.sql.analyzer.CTASAnalyzer.transformCTASStmt(CTASAnalyzer.java:44) ~[classes/:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitCreateTableAsSelectStatement(Analyzer.java:80) ~[classes/:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitCreateTableAsSelectStatement(Analyzer.java:46) ~[classes/:?]
        at com.starrocks.analysis.CreateTableAsSelectStmt.accept(CreateTableAsSelectStmt.java:118) ~[classes/:?]
        at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:63) ~[classes/:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.analyze(Analyzer.java:49) ~[classes/:?]
        at com.starrocks.sql.analyzer.Analyzer.analyze(Analyzer.java:43) ~[classes/:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:56) ~[classes/:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:312) ~[classes/:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:284) ~[classes/:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:431) ~[classes/:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:667) ~[classes/:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[classes/:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_202]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_202]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_202]
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.2

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5500

## Problem Summary(Required) ：

This because interval is expr and can not cast to Long
then catch exception use old got an strange error
we should log some error for new parser

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 4 / 7 (57.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/MultiRangePartitionDesc.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|5|80.00%|

File: fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java [hunk 1]
```diff
@@ -264,6 +264,7 @@ private void handleQuery() {
             } catch (ParsingException parsingException) {
                 throw new AnalysisException(parsingException.getMessage());
             } catch (Exception e) {
+                LOG.warn("failed to parse stmt with new parser", e);
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java:267]: Adding warn logs here may cause a large number of warn logs to be printed. Any statement not supported by the new Parser will be printed, causing confusion for user

**Review by HangyuanLiu — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 4 / 6 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/MultiRangePartitionDesc.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|5|80.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 2]
```diff
@@ -386,11 +386,17 @@ public ParseNode visitSingleRangePartition(StarRocksParser.SingleRangePartitionC
     public ParseNode visitMultiRangePartition(StarRocksParser.MultiRangePartitionContext context) {
         if (context.interval() != null) {
             IntervalLiteral intervalLiteral = (IntervalLiteral) visit(context.interval());
-
+            Expr expr = intervalLiteral.getValue();
+            String intervalVal;
+            if (expr instanceof IntLiteral) {
+                intervalVal = ((IntLiteral) expr).getStringValue();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:392]: Why not return long type directly?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 4 / 6 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/MultiRangePartitionDesc.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|5|80.00%|

**Review by wyb — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**